### ISSUE TITLE
Conversion upload error

### ISSUE THREAD

**Issue by psjbeisler** [LycheeOrg/Lychee-Laravel#396]
### Detailed description of the problem
Duplicate of #186
Failure to upload certain photos due to a sql conversion error (specifically in PSQL)
Think i have it tracked to ISO Speed Ratings: on some photos having 3 values and not 1

`exiv2 -PE *|grep ISO`

Failure:
`IMG_20150115_224548_1.jpg  Exif.Photo.ISOSpeedRatings                   Short       3  640`
Success:
`IMG_20190209_170313.jpg  Exif.Photo.ISOSpeedRatings                   Short       1  725`

### Steps to reproduce the issue
Apparently `Exif.Photo.ISOSpeedRatings` is a r/o value, but if I clear _all_ the exif data tho, I can upload without an issue.
1) Find a photo matching description
2) upload and fail
3) use exiftool/exiv2 and clear EXIF metadata
4) upload succeeds

**Screenshots**
https://imgur.com/a/A7bE8e7

### Output of the diagnostics
```
2019-04-12 04:23:16 -- notice -- App\Image\ImagickHandler::crop -- 87 -- Saving thumb to /var/www/html/Lychee-Laravel/public/uploads/thumb/452004e6a9a243d714a1172054b6e757@2x.jpeg
2019-04-12 04:23:15 -- notice -- App\ModelFunctions\PhotoFunctions::createThumb -- 101 -- Photo URL is 48006823d3260010ac55b267305f6e0b.jpg
2019-04-12 04:23:15 -- notice -- App\Image\ImagickHandler::crop -- 87 -- Saving thumb to /var/www/html/Lychee-Laravel/public/uploads/thumb/452004e6a9a243d714a1172054b6e757.jpeg
2019-04-12 04:23:14 -- notice -- App\ModelFunctions\PhotoFunctions::createThumb -- 101 -- Photo URL is 452004e6a9a243d714a1172054b6e757.jpg
2019-04-12 04:23:14 -- error -- App\ModelFunctions\PhotoFunctions::save -- 480 -- Something went wrong, error 0, Array to string conversion (SQL: insert into "photos" ("id", "title", "url", "description", "tags", "width", "height", "type", "size", "iso", "aperture", "make", "model", "lens", "shutter", "focal", "takestamp", "latitude", "longitude", "altitude", "public", "owner_id", "star", "checksum", "album_id", "thumb2x", "thumbUrl", "medium", "small", "updated_at", "created_at") values (15550429934893, IMG_20150115_224548_1, 1bf70c4a3430ad28033aed2b39a28611.jpg, , , 2592, 1944, image/jpeg, 1.9 MB, 640, f/2.8, Samsung, Galaxy Nexus, , 59994/1000000 s, 3.4 mm, 2015-01-15 22:45:48, , , , 0, 0, 0, 99868c5e41c114e428f24004ff8c2811fd53415a, 15550429810009, 1, 1bf70c4a3430ad28033aed2b39a28611.jpeg, 1440x1080, 480x360, 2019-04-12 04:23:14, 2019-04-12 04:23:14) returning "id")
2019-04-12 04:23:14 -- notice -- App\Image\ImagickHandler::scale -- 50 -- Saving thumb to /var/www/html/Lychee-Laravel/public/uploads/small/1bf70c4a3430ad28033aed2b39a28611.jpg
2019-04-12 04:23:14 -- notice -- App\Image\ImagickHandler::scale -- 50 -- Saving thumb to /var/www/html/Lychee-Laravel/public/uploads/medium/1bf70c4a3430ad28033aed2b39a28611.jpg
2019-04-12 04:23:14 -- notice -- App\ModelFunctions\PhotoFunctions::add -- 345 -- 1bf70c4a3430ad28033aed2b39a28611.jpeg
2019-04-12 04:23:14 -- notice -- App\Image\ImagickHandler::crop -- 87 -- Saving thumb to /var/www/html/Lychee-Laravel/public/uploads/thumb/1bf70c4a3430ad28033aed2b39a28611@2x.jpeg
2019-04-12 04:23:13 -- notice -- App\Image\ImagickHandler::crop -- 87 -- Saving thumb to /var/www/html/Lychee-Laravel/public/uploads/thumb/1bf70c4a3430ad28033aed2b39a28611.jpeg
2019-04-12 04:23:13 -- notice -- App\ModelFunctions\PhotoFunctions::createThumb -- 101 -- Photo URL is 1bf70c4a3430ad28033aed2b39a28611.jpg
2019-04-12 04:23:11 -- error -- App\ModelFunctions\PhotoFunctions::save -- 480 -- Something went wrong, error 0, Array to string conversion (SQL: insert into "photos" ("id", "title", "url", "description", "tags", "width", "height", "type", "size", "iso", "aperture", "make", "model", "lens", "shutter", "focal", "takestamp", "latitude", "longitude", "altitude", "public", "owner_id", "star", "checksum", "album_id", "thumb2x", "thumbUrl", "medium", "small", "updated_at", "created_at") values (15550429906233, IMG_20140725_190218, 3a24b082bf9387f3cc56398976bbae6f.jpg, , , 1944, 2592, image/jpeg, 1016.5 KB, 100, f/2.8, Samsung, Galaxy Nexus, , 59994/1000000 s, 3.4 mm, 2014-07-25 19:02:18, , , , 0, 0, 0, ec2674c07f01e86dff48cbe82ef6ddbfc79ede8c, 15550429810009, 1, 3a24b082bf9387f3cc56398976bbae6f.jpeg, 810x1080, 270x360, 2019-04-12 04:23:11, 2019-04-12 04:23:11) returning "id")
2019-04-12 04:23:11 -- notice -- App\Image\ImagickHandler::scale -- 50 -- Saving thumb to /var/www/html/Lychee-Laravel/public/uploads/small/3a24b082bf9387f3cc56398976bbae6f.jpg
2019-04-12 04:23:11 -- notice -- App\Image\ImagickHandler::scale -- 50 -- Saving thumb to /var/www/html/Lychee-Laravel/public/uploads/medium/3a24b082bf9387f3cc56398976bbae6f.jpg
2019-04-12 04:23:11 -- notice -- App\ModelFunctions\PhotoFunctions::add -- 345 -- 3a24b082bf9387f3cc56398976bbae6f.jpeg
2019-04-12 04:23:11 -- notice -- App\Image\ImagickHandler::crop -- 87 -- Saving thumb to /var/www/html/Lychee-Laravel/public/uploads/thumb/3a24b082bf9387f3cc56398976bbae6f@2x.jpeg
2019-04-12 04:23:10 -- notice -- App\Image\ImagickHandler::crop -- 87 -- Saving thumb to /var/www/html/Lychee-Laravel/public/uploads/thumb/3a24b082bf9387f3cc56398976bbae6f.jpeg
2019-04-12 04:23:10 -- notice -- App\ModelFunctions\PhotoFunctions::createThumb -- 101 -- Photo URL is 3a24b082bf9387f3cc56398976bbae6f.jpg
2019-04-12 04:23:10 -- error -- App\ModelFunctions\PhotoFunctions::save -- 480 -- Something went wrong, error 0, Array to string conversion (SQL: insert into "photos" ("id", "title", "url", "description", "tags", "width", "height", "type", "size", "iso", "aperture", "make", "model", "lens", "shutter", "focal", "takestamp", "latitude", "longitude", "altitude", "public", "owner_id", "star", "checksum", "album_id", "thumb2x", "thumbUrl", "medium", "small", "updated_at", "created_at") values (15550429892454, IMG_20140416_230210, a1e855d97f64461460843415b30213e1.jpg, , , 2592, 1944, image/jpeg, 878.6 KB, 640, f/2.8, Samsung, Galaxy Nexus, , 58322/1000000 s, 3.4 mm, 2014-04-16 23:02:10, , , , 0, 0, 0, fc27e250541400d0b93f31c46e30d4664d6eed14, 15550429810009, 1, a1e855d97f64461460843415b30213e1.jpeg, 1440x1080, 480x360, 2019-04-12 04:23:10, 2019-04-12 04:23:10) returning "id")
2019-04-12 04:23:10 -- notice -- App\Image\ImagickHandler::scale -- 50 -- Saving thumb to /var/www/html/Lychee-Laravel/public/uploads/small/a1e855d97f64461460843415b30213e1.jpg
2019-04-12 04:23:10 -- notice -- App\Image\ImagickHandler::scale -- 50 -- Saving thumb to /var/www/html/Lychee-Laravel/public/uploads/medium/a1e855d97f64461460843415b30213e1.jpg
2019-04-12 04:23:10 -- notice -- App\ModelFunctions\PhotoFunctions::add -- 345 -- a1e855d97f64461460843415b30213e1.jpeg
2019-04-12 04:23:10 -- notice -- App\Image\ImagickHandler::crop -- 87 -- Saving thumb to /var/www/html/Lychee-Laravel/public/uploads/thumb/a1e855d97f64461460843415b30213e1@2x.jpeg
2019-04-12 04:23:09 -- notice -- App\Image\ImagickHandler::crop -- 87 -- Saving thumb to /var/www/html/Lychee-Laravel/public/uploads/thumb/a1e855d97f64461460843415b30213e1.jpeg
2019-04-12 04:23:09 -- notice -- App\ModelFunctions\PhotoFunctions::createThumb -- 101 -- Photo URL is a1e855d97f64461460843415b30213e1.jpg
```

### Browser and system
Firefox / Arch Linux
Firefox / Android

**Comment by psjbeisler** [comment]
The only thing Id really be missing here is knowing the "Taken date" for some photos, by clearing all the metadata. Device and such are still nice to have...

**Comment by ildyria** [comment]
That is an interesting case. Thank you for tracing out where this error came from. :)

**Comment by tmp-hallenser** [comment]
@psjbeisler Do you have a sample photo to test to current PR to support the exiftool?

**Comment by psjbeisler** [comment]
https://psjb.me/bad_example.jpg

**Comment by tmp-hallenser** [comment]
Thanks! Will be fixed once #401 is merged.
